### ISSUE TITLE
Multisig Updates - confirmed requests are not removed from storage

### ISSUE THREAD

**Issue by mattlockyer** [near/core-contracts#67]
When confirming a request on the new multisig contract https://github.com/near/core-contracts/pull/66 requests don't appear to be removed properly.

Steps to reproduce:
1. deploy the contract as 1/1 confirmations multisig
2. create a new request to change confirmations to 1 (trivial request)
3. confirm request
4. **request still in storage** via `list_request_ids` i.e. request id is returned in Vec<u32>
5. delete_request produces error [here](https://github.com/near/core-contracts/blob/c4fdb2285ab1afc13e3637613fe62126bd19244c/multisig/src/lib.rs#L137)
6. confirming again produces error [here](https://github.com/near/core-contracts/blob/c4fdb2285ab1afc13e3637613fe62126bd19244c/multisig/src/lib.rs#L239)
7. `get_confirmations` produces error [here](https://github.com/near/core-contracts/blob/c4fdb2285ab1afc13e3637613fe62126bd19244c/multisig/src/lib.rs#L273)

Based on (7) it seems like the request is removed from confirmations, but based on (4) not removed from requests.

Currently looking into this. @ilblackdragon might be able to spot why a bit faster.

**Comment by mattlockyer** [comment]
Added a branch you can try [here](https://github.com/near/core-contracts/blob/86ee00d77d1019f9556a1fc4886cfb0d74e2beb0/multisig/src/lib.rs#L248) and updated the confirm method to check if the request `is_some()` after calling `execute_request()`

Not sure why the requests are not being removed.

I still get a true value from `on_transaction_completed` even if I modify it to the following:
```
...
if transaction_index + 1 == request.len() {
    // Request is finished and can be removed.
    self.requests.remove(&request_id);
    true
} else {
    self.execute_request(request_id, transaction_index + 1, request)
}
```

**Comment by mattlockyer** [comment]
The issue is worse than just the requests not being removed. They are not being executed at all.
https://explorer.testnet.near.org/accounts/dev-1591805379333

**Comment by mattlockyer** [comment]
I believe I've resolved this issue in [debug-multisig](https://github.com/near/core-contracts/blob/debug-multisig/multisig/src/lib.rs)

The main issue was the trait was `next_transaction_index` but implemented as `transaction_index`. Extremely hard to track down, at least for me.

I've updated a few things so that confirmations are not removed before the requests.

I'm going to continue e2e testing with my app.
